### PR TITLE
Layout Item 3D Map Widget change

### PR DESCRIPTION
**Pull request by JanCaha**
## Description

This PR allows `QgsLayout3DMapWidget` to copy 3D Map Settings and Camera Pose from `3D View` that are not currently open as QGIS `3D Map View`. The widget will now list all existing `3D Views` in menus (in contrast to only listing opened 3D Views Widgets) for copying 3D scenes settings and camera positions.

This should be useful when working on Layout where user wants to add 3D Maps, which are not currently open (and in fact do not need to open).

### CODE REVIEW

**Review by uclaros — COMMENTED** [review]
Nice addition! This is very useful now that 3d views persist after they are closed.
Check the couple of suggestions below

**Comment by JanCaha** [comment]
Thanks for the suggestions :) I tried to include them, but I took slightly different approach. Trying to construct `Qgs3DMapSettings` and `QgsCameraPose` from the `QDomElement` in the `Widget`. That should avoid the need to construct the whole scene. 

The main reason why I tried this way instead of what you suggested is that, for some `Qgs3DMapSettings` the `QgisApp`  is needed, which is not available in `QgsLayoutItem3DMap` and I did not really want to to bring there.

Another slight issue is that the settings of camera are stored as `QgsCameraController` but actually  `QgsCameraPose` si needed. So I had to parse it myself to avoid creating `QgsCameraController`, which requires `Qgs3DMapScene` ... 

Maybe the design needs some more thinking ...

**Comment by nyalldawson** [comment]
@uclaros how's this looking to you now?

**Comment by uclaros** [comment]
Couldn't we just get `selectionColor` and `backgroundColor` from the project instance and not require QgisApp?

**Comment by JanCaha** [comment]
@uclaros I did some changes to the code based on the discussion that we had. 

The 3dMapSettings is not a problem, that can be set directly from XML.

The CameraPose is a little bit trickier, because the XML only store CameraController, which is slightly different object. The  CameraPose is read from the XML on QgsLayoutItem3dMap, but it is read from CameraController, hence the xml parsing in `QgsLayoutItem3DMap::setCameraPoseFromXml()`. All other approaches would require touching other classes or initiating the scene, which I wanted to avoid.

Another minor addition is that, if the 3DView is currently open, the CameraPose is obtained directly from 3DView. This should likely be faster and also captures the true position of camera, even if it was not yet saved into XML.

Any ideas for changes?

**Review by uclaros — COMMENTED** [review]
Sorry for the late review @JanCaha !
While this looks good to me, I am getting a crash if I try to re-copy settings from existing 3d view...

**Comment by JanCaha** [comment]
@uclaros hopefully this will ok ;-)

File: src/3d/qgslayoutitem3dmap.cpp [hunk 1]
```diff
@@ -307,3 +311,36 @@ void QgsLayoutItem3DMap::setCameraPose( const QgsCameraPose &pose )
...
+  pose.setPitchAngle( elem.attribute( "pitch" ).toDouble() );
+
+  setCameraPose( pose );
+}
+
+void QgsLayoutItem3DMap::loadProjectSettings()
+{
+  mSettings->setSelectionColor( QgsProject::instance()->selectionColor() );
+  mSettings->setBackgroundColor( QgsProject::instance()->backgroundColor() );
+  mSettings->setOutputDpi( QGuiApplication::primaryScreen()->logicalDotsPerInch() );
```
Comment (nyalldawson) [inline src/3d/qgslayoutitem3dmap.cpp:345]:
```suggestion
  QgsProject* project = layout() && layout()->project();
  if ( !project )
    return;
  mSettings->setSelectionColor( project->selectionColor() );
  mSettings->setBackgroundColor( project->backgroundColor() );
```

**Comment by JanCaha** [comment]
Should be all fixed @nyalldawson, as suggested :)

**Pull request closed without merging**